airgen/src/lib.rs

```diff
@@ -118,23 +116,21 @@ impl<'a, T: FieldElement> ASMPILConverter<'a, T> {
 
         self.submachines = input.submachines;
 
-        // machines should only have constraints, functions and external instructions at this point
-        assert!(input
-            .instructions
-            .iter()
-            .all(|i| matches!(i.instruction.body, InstructionBody::External(..))));
+        // machines should only have constraints, operations and links at this point
+        assert!(input.instructions.is_empty());
```

Comment:
```suggestion
        assert!(input.instructions.is_empty());
        assert!(input.function_definitions().next().is_none());
```

Reply:
This is already taken care of by `input.callable.is_only_operations()` down below